app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,46 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [ :show, :edit, :update, :destroy ]
+
+  def index
+    @posts = Post.all
+  end
+
+  def show
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to @post, notice: "Post was successfully created."
```

Comment:
インスタンスで遷移させる方法もいいのですが、すべてをこの形式で書ききれなくなることを考えると、URLヘルパーつかう形に統一しておくとよいです。
```suggestion
      redirect_to post_path(@post), notice: "Post was successfully created."
```